client/grpc/tmservice/service.go

```diff
@@ -192,16 +216,38 @@ func (s queryServer) GetNodeInfo(ctx context.Context, req *GetNodeInfoRequest) (
 	return &resp, nil
 }
 
+func (s queryServer) ABCIQuery(ctx context.Context, req *ABCIQueryRequest) (*ABCIQueryResponse, error) {
+	if s.queryFn == nil {
+		return nil, status.Error(codes.Internal, "ABCI Query handler undefined")
+	}
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	if path := baseapp.SplitABCIQueryPath(req.Path); len(path) > 0 {
```

Comment:
should we error here in the else case? i.e. if `len(path) == 0`

Reply:
We can. It doesn't hurt, but `Baseapp#Query` already does this for us.